samples/subsys/usb/console/nrf52840_pca10059.overlay

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
```

Comment:
Well if the following snipped is not added to the device tree of the soc/board, the sample will not build:

```
                usb_cdc: virtualcom {
                        compatible = "nordic,nrf-usbd";
                        label = "CDC_ACM";
                };
```

On the other hand the sample will build but won't work without this overlay.

Alternatively I don't know if we could easily use symlink so that at least the same file is used.

Reply:
We have some boards with default console over USB (tinytile), now we need to put those overlays to every sample?

Reply:
Before it was in Kconfig, everything was much better

Reply:
But this does not affect this PR

Reply:
> But this does not affect this PR

I didn't feel like separating this last commit from the others, as it's basically how I tested it. Now if it requires some discussion, I can just drop it from this PR and move it to a new one.

Reply:
Leave it here, we might fix this later.

Reply:
The same overlay can also be used for usb-kw24d512 board. Just wonder what virtualcom is and where it comes from...

Reply:
Ok. As changing that seems controversial, I have just dropped the nrf52840 dongle related commits. I'll do a separate pull request once this PR is merged.